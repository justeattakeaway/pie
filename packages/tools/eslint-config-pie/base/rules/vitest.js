const vitest = require('eslint-plugin-vitest');

module.exports = {
    plugins: [
        'vitest',
    ],
    rules: {
        ...vitest.configs.recommended.rules,
    },
    globals: {
        ...vitest.environments.env.globals,
    },
};
