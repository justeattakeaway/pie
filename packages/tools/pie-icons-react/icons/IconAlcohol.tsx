import * as React from "react";
import { RegularIconProps } from "../types";
// @ts-ignore
import { getSvgProps } from './configs-react';
const IconAlcohol = (props: RegularIconProps) => {
  const {
    className,
    iconSize,
    width,
    height,
    ...remainingProps
  } = props;
  const moreProps = getSvgProps("pie-icon pie-icon--alcohol", className, iconSize, "IconAlcohol");
  const allProps = {
    ...remainingProps,
    ...moreProps
  };
  return <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"    {...allProps}><path fill="#242E30" d="M6.4 5.28V2.576c0-.419-.17-.82-.472-1.115A1.624 1.624 0 0 0 4.791 1h-.68c-.426 0-.835.166-1.137.462a1.561 1.561 0 0 0-.47 1.115v2.702a2.11 2.11 0 0 1-.534 1.45 4.613 4.613 0 0 0-.964 3.055v3.603c-.015.401.132.792.409 1.087.277.295.661.471 1.07.49h3.933a1.566 1.566 0 0 0 1.07-.49c.277-.295.424-.686.41-1.087V9.784a4.611 4.611 0 0 0-.92-3.054A2.112 2.112 0 0 1 6.4 5.28ZM3.881 2.576c0-.06.024-.117.067-.16a.232.232 0 0 1 .162-.066h.68c.061 0 .12.024.163.066.043.043.067.1.067.16v1.126h-1.14V2.577Zm1.14 2.477v.225a3.426 3.426 0 0 0 .77 2.19c.168.222.313.46.433.711H2.678c.12-.25.265-.489.432-.712a3.426 3.426 0 0 0 .772-2.189v-.225h1.14Zm1.497 8.333a.257.257 0 0 1-.101.226H2.485a.262.262 0 0 1-.101-.226V9.56H6.51v.225l.009 3.603ZM8.605 3.081v4c0 .713.249 1.405.705 1.96a3.198 3.198 0 0 0 1.803 1.094v3.514h-1.46V15h4.3v-1.351h-1.461v-3.514a3.198 3.198 0 0 0 1.803-1.093A3.088 3.088 0 0 0 15 7.082v-4H8.605Zm5.035 4c0 .478-.194.936-.538 1.274a1.856 1.856 0 0 1-1.3.528c-.487 0-.954-.19-1.3-.528a1.784 1.784 0 0 1-.537-1.274V4.432h3.675v2.65Z" /></svg>;
};
export default IconAlcohol;