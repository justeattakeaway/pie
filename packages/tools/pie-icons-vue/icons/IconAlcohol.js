import _mergeJSXProps from '@vue/babel-helper-vue-jsx-merge-props';

export default {
    name: 'IconAlcohol',
    props: {},
    functional: true,
    render (h, ctx) {
        const attrs = ctx.data.attrs || {};
        ctx.data.attrs = attrs;
        return h('svg', _mergeJSXProps([{
            attrs: {
                xmlns: 'http://www.w3.org/2000/svg',
                viewBox: '0 0 16 16'
            },
            class: 'c-pieIcon c-pieIcon--alcohol'
        }, ctx.data]), [h('path', {
            attrs: {
                d: 'M6.39939 5.27928V5.05405V2.57658C6.39939 2.15844 6.22997 1.75743 5.92841 1.46177C5.62685 1.1661 5.21784 1 4.79137 1H4.1114C3.68493 1 3.27592 1.1661 2.97436 1.46177C2.6728 1.75743 2.50338 2.15844 2.50338 2.57658V5.05405V5.27928C2.51618 5.81097 2.32613 6.3282 1.97044 6.72973C1.2928 7.60497 0.951213 8.68624 1.00563 9.78378V13.3874C0.990656 13.7883 1.13765 14.1787 1.41466 14.4739C1.69167 14.7691 2.07631 14.9452 2.48501 14.964H6.41777C6.82646 14.9452 7.2111 14.7691 7.48811 14.4739C7.76512 14.1787 7.91212 13.7883 7.89714 13.3874V9.78378C7.96473 8.69172 7.63957 7.61099 6.97828 6.72973C6.60592 6.33496 6.39919 5.81698 6.39939 5.27928ZM3.88169 2.57658C3.88169 2.51684 3.90589 2.45956 3.94897 2.41732C3.99205 2.37508 4.05048 2.35135 4.1114 2.35135H4.79137C4.85229 2.35135 4.91072 2.37508 4.9538 2.41732C4.99688 2.45956 5.02108 2.51684 5.02108 2.57658V3.7027H3.88169V2.57658ZM5.02108 5.05405V5.27928C5.0149 6.07417 5.28739 6.84702 5.79293 7.46847C5.96021 7.69133 6.10486 7.9297 6.2248 8.18018H2.67797C2.79791 7.9297 2.94256 7.69133 3.10984 7.46847C3.61538 6.84702 3.88787 6.07417 3.88169 5.27928V5.05405H5.02108ZM6.51884 13.3874C6.52216 13.4302 6.51467 13.4731 6.49704 13.5124C6.47941 13.5517 6.45218 13.5861 6.41777 13.6126H2.48501C2.45059 13.5861 2.42336 13.5517 2.40573 13.5124C2.3881 13.4731 2.38062 13.4302 2.38393 13.3874V9.78378C2.38393 9.7027 2.38393 9.63063 2.38393 9.55856H6.50965C6.50965 9.63063 6.50965 9.7027 6.50965 9.78378L6.51884 13.3874ZM8.60467 3.08108V7.08108C8.60471 7.79434 8.85352 8.48615 9.30981 9.04167C9.7661 9.59719 10.4024 9.98303 11.1132 10.1351V13.6486H9.65218V15H13.9525V13.6486H12.4915V10.1351C13.2022 9.98303 13.8386 9.59719 14.2949 9.04167C14.7512 8.48615 15 7.79434 15 7.08108V3.08108H8.60467ZM13.6401 7.08108C13.6401 7.55895 13.4465 8.01724 13.1018 8.35515C12.7572 8.69305 12.2897 8.88288 11.8023 8.88288C11.3149 8.88288 10.8475 8.69305 10.5029 8.35515C10.1582 8.01724 9.9646 7.55895 9.9646 7.08108V4.43243H13.6401V7.08108Z',
                fill: '#242E30'
            }
        })]);
    }
};
