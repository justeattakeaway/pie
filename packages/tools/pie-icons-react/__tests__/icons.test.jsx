import React from 'react';
import renderer from 'react-test-renderer';
import { describe, expect, it } from 'vitest';

import IconAlcohol from '../icons/IconAlcohol';

describe('Icon', () => {
    it('renders correctly without className prop', () => {
        const tree = renderer
        .create(<IconAlcohol />)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with className prop', () => {
        const tree = renderer
        .create(<IconAlcohol className="additional-styling" />)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
