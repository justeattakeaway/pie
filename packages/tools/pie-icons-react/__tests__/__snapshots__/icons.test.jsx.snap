// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon > renders correctly with className prop 1`] = `
<svg
  className="c-pieIcon c-pieIcon--alcohol additional-styling"
  fill="currentColor"
  focusable="false"
  height={16}
  role="presentation"
  viewBox="0 0 16 16"
  width={16}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M6.4 5.28V2.576c0-.419-.17-.82-.472-1.115A1.624 1.624 0 0 0 4.791 1h-.68c-.426 0-.835.166-1.137.462a1.561 1.561 0 0 0-.47 1.115v2.702a2.11 2.11 0 0 1-.534 1.45 4.613 4.613 0 0 0-.964 3.055v3.603c-.015.401.132.792.409 1.087.277.295.661.471 1.07.49h3.933a1.566 1.566 0 0 0 1.07-.49c.277-.295.424-.686.41-1.087V9.784a4.611 4.611 0 0 0-.92-3.054A2.112 2.112 0 0 1 6.4 5.28ZM3.881 2.576c0-.06.024-.117.067-.16a.232.232 0 0 1 .162-.066h.68c.061 0 .12.024.163.066.043.043.067.1.067.16v1.126h-1.14V2.577Zm1.14 2.477v.225a3.426 3.426 0 0 0 .77 2.19c.168.222.313.46.433.711H2.678c.12-.25.265-.489.432-.712a3.426 3.426 0 0 0 .772-2.189v-.225h1.14Zm1.497 8.333a.257.257 0 0 1-.101.226H2.485a.262.262 0 0 1-.101-.226V9.56H6.51v.225l.009 3.603ZM8.605 3.081v4c0 .713.249 1.405.705 1.96a3.198 3.198 0 0 0 1.803 1.094v3.514h-1.46V15h4.3v-1.351h-1.461v-3.514a3.198 3.198 0 0 0 1.803-1.093A3.088 3.088 0 0 0 15 7.082v-4H8.605Zm5.035 4c0 .478-.194.936-.538 1.274a1.856 1.856 0 0 1-1.3.528c-.487 0-.954-.19-1.3-.528a1.784 1.784 0 0 1-.537-1.274V4.432h3.675v2.65Z"
  />
</svg>
`;

exports[`Icon > renders correctly with className prop 2`] = `
<svg
  className="c-pieIcon c-pieIcon--alcohol-large additional-styling"
  fill="currentColor"
  focusable="false"
  height={32}
  role="presentation"
  viewBox="0 0 32 32"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12.386 10.811V5.5a2.625 2.625 0 0 0-2.625-2.625H8.466A2.625 2.625 0 0 0 5.841 5.5v5.311a4.655 4.655 0 0 1-1.085 3.01 8.671 8.671 0 0 0-1.75 5.679v7a2.52 2.52 0 0 0 2.38 2.625h7.499a2.511 2.511 0 0 0 2.371-2.625v-7a8.75 8.75 0 0 0-1.75-5.679 4.612 4.612 0 0 1-1.12-3.01ZM7.591 5.5a.875.875 0 0 1 .875-.875H9.77a.875.875 0 0 1 .875.875v2.625H7.591V5.5Zm-1.4 9.31a6.248 6.248 0 0 0 1.4-3.999v-.936h3.045v.936a6.248 6.248 0 0 0 1.4 3.999c.27.38.516.778.735 1.19H5.5c.184-.429.438-.805.691-1.19ZM13.49 26.5c0 .473-.28.875-.622.875H5.37c-.341 0-.63-.402-.63-.875v-7c-.005-.586.042-1.172.14-1.75h8.496c.098.578.145 1.164.14 1.75l-.026 7Zm15.286-12.329V6.918H17.453v7.253a5.679 5.679 0 0 0 4.786 5.591v7.613h-3.22v1.75h8.199v-1.75h-3.23v-7.613a5.67 5.67 0 0 0 4.787-5.59Zm-1.75-5.503v2.528c-2.406.7-3.325.403-4.375.044a6.825 6.825 0 0 0-3.5-.42V8.668h7.875Zm-7.822 5.503V12.58a5.127 5.127 0 0 1 2.896.315 6.93 6.93 0 0 0 2.45.481 10.232 10.232 0 0 0 2.476-.359v1.155a3.911 3.911 0 1 1-7.822 0Z"
  />
</svg>
`;

exports[`Icon > renders correctly without className prop 1`] = `
<svg
  className="c-pieIcon c-pieIcon--alcohol"
  fill="currentColor"
  focusable="false"
  height={16}
  role="presentation"
  viewBox="0 0 16 16"
  width={16}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M6.4 5.28V2.576c0-.419-.17-.82-.472-1.115A1.624 1.624 0 0 0 4.791 1h-.68c-.426 0-.835.166-1.137.462a1.561 1.561 0 0 0-.47 1.115v2.702a2.11 2.11 0 0 1-.534 1.45 4.613 4.613 0 0 0-.964 3.055v3.603c-.015.401.132.792.409 1.087.277.295.661.471 1.07.49h3.933a1.566 1.566 0 0 0 1.07-.49c.277-.295.424-.686.41-1.087V9.784a4.611 4.611 0 0 0-.92-3.054A2.112 2.112 0 0 1 6.4 5.28ZM3.881 2.576c0-.06.024-.117.067-.16a.232.232 0 0 1 .162-.066h.68c.061 0 .12.024.163.066.043.043.067.1.067.16v1.126h-1.14V2.577Zm1.14 2.477v.225a3.426 3.426 0 0 0 .77 2.19c.168.222.313.46.433.711H2.678c.12-.25.265-.489.432-.712a3.426 3.426 0 0 0 .772-2.189v-.225h1.14Zm1.497 8.333a.257.257 0 0 1-.101.226H2.485a.262.262 0 0 1-.101-.226V9.56H6.51v.225l.009 3.603ZM8.605 3.081v4c0 .713.249 1.405.705 1.96a3.198 3.198 0 0 0 1.803 1.094v3.514h-1.46V15h4.3v-1.351h-1.461v-3.514a3.198 3.198 0 0 0 1.803-1.093A3.088 3.088 0 0 0 15 7.082v-4H8.605Zm5.035 4c0 .478-.194.936-.538 1.274a1.856 1.856 0 0 1-1.3.528c-.487 0-.954-.19-1.3-.528a1.784 1.784 0 0 1-.537-1.274V4.432h3.675v2.65Z"
  />
</svg>
`;

exports[`Icon > renders correctly without className prop 2`] = `
<svg
  className="c-pieIcon c-pieIcon--alcohol-large"
  fill="currentColor"
  focusable="false"
  height={32}
  role="presentation"
  viewBox="0 0 32 32"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12.386 10.811V5.5a2.625 2.625 0 0 0-2.625-2.625H8.466A2.625 2.625 0 0 0 5.841 5.5v5.311a4.655 4.655 0 0 1-1.085 3.01 8.671 8.671 0 0 0-1.75 5.679v7a2.52 2.52 0 0 0 2.38 2.625h7.499a2.511 2.511 0 0 0 2.371-2.625v-7a8.75 8.75 0 0 0-1.75-5.679 4.612 4.612 0 0 1-1.12-3.01ZM7.591 5.5a.875.875 0 0 1 .875-.875H9.77a.875.875 0 0 1 .875.875v2.625H7.591V5.5Zm-1.4 9.31a6.248 6.248 0 0 0 1.4-3.999v-.936h3.045v.936a6.248 6.248 0 0 0 1.4 3.999c.27.38.516.778.735 1.19H5.5c.184-.429.438-.805.691-1.19ZM13.49 26.5c0 .473-.28.875-.622.875H5.37c-.341 0-.63-.402-.63-.875v-7c-.005-.586.042-1.172.14-1.75h8.496c.098.578.145 1.164.14 1.75l-.026 7Zm15.286-12.329V6.918H17.453v7.253a5.679 5.679 0 0 0 4.786 5.591v7.613h-3.22v1.75h8.199v-1.75h-3.23v-7.613a5.67 5.67 0 0 0 4.787-5.59Zm-1.75-5.503v2.528c-2.406.7-3.325.403-4.375.044a6.825 6.825 0 0 0-3.5-.42V8.668h7.875Zm-7.822 5.503V12.58a5.127 5.127 0 0 1 2.896.315 6.93 6.93 0 0 0 2.45.481 10.232 10.232 0 0 0 2.476-.359v1.155a3.911 3.911 0 1 1-7.822 0Z"
  />
</svg>
`;
